### DISCUSSION TITLE
Tuning CP-SAT parameters and search strategy

### DISCUSSION THREAD

**Discussion by mkrechetov** [google/or-tools#3149] [CP-SAT questions] [answered]
Hello!

I am solving an optimization problem with cp-sat and the problem consists of multiple linear equality/inequality constraints under OnlyEnforceIf conditions. With default solver parameters it takes, say, 400 seconds including 31 seconds for pre-solve. I have tried tweaking most of the parameters from sat_parameters.proto (https://github.com/google/or-tools/blob/604587eec79d9c84dd0104b37e82e08b5e656138/ortools/sat/sat_parameters.proto) and I was not able to reduce the running time.

Not surprisingly, when I tried 'solver.parameters.randomize_search = True' with different 'solver.parameters.random_seed' values, I have observed the crucial running time fluctuations from 39 seconds (including 31 seconds presolve) and up to tens of minutes of total running time.

The question is how can I extract the solver's (as well as search) settings for 'successful' seed values? I want to get an intuition on how to tune the solver for my problem without bruteforcing 'random_seed' values. It would be great if it is possible to extract the search order corresponding to a random_seed and then set it manually for future runs.

Thanks in advance!

**Comment by lperron** [comment]
Are you running with multiple workers ?

**Reply by mkrechetov** [reply]
No, I set solver.parameters.num_search_workers = 1

**Reply by mkrechetov** [reply]
All parameters, except num_search_workers, time limit, randomize_search and random_seed, are default

**Comment by lperron** [comment, marked as answer]
Just try

   log_search_progress = True
   num_search_workers = 8

And nothing else.

Is you have more cores, you can bump 8 to to 12 or 16.


Le lun. 28 févr. 2022, 04:01, mkrechetov ***@***.***> a
écrit :
